addons/project/tests/test_burndown_chart.py

```diff
@@ -354,6 +354,37 @@ def test_burndown_chart(self):
         }
         self.check_read_group_is_closed_results(complex_domain, complex_domain_expected_dict)
 
+    def test_burndown_chart_stage_report(self):
+        burndown_chart_domain = [('project_id', '!=', False)]
+        project_domain = [('project_id', '=', self.project_1.id)]
+
+        with freeze_time('%s-01-01' % (self.current_year)):
+            self.task_a.write({'project_id': self.project_1.id})
+            self.task_a.write({'stage_id': self.stage_4.id})
+            self.env.cr.flush()
+
+        project_expected_dict = {
+            ('January %s' % (self.current_year - 1), self.stage_1.id): 1,
+            ('February %s' % (self.current_year - 1), self.stage_2.id): 1,
+            ('March %s' % (self.current_year - 1), self.stage_3.id): 1,
+            ('April %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('May %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('June %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('July %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('August %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('September %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('October %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('November %s' % (self.current_year - 1), self.stage_4.id): 1,
+            ('December %s' % (self.current_year - 1), self.stage_4.id): 1,
+        }
+        months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October',
+                  'November', 'December']
+        current_month = datetime.now().month
+        for i in range(current_month):
+            month_key = f"{months[i]} {self.current_year}"
+            project_expected_dict[(month_key, self.stage_4.id)] = 2
```

Comment:
why do we create the ``project_expected_dict`` and the immediately change it here without doing anything else with it (of course before we use it in line 386)? can't we immediately make the changes in lines 366 to 378?

Reply:
alright! I understand why we are doing it, but I am finding the way we go by doing it a bit convoluted. Wouldn't something like this be a bit easier to understand and less hard-coded:
```py
last_year = self.current_year - 1
months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October',
          'November', 'December']
current_month = datetime.now().month
project_expected_dict = {
    ('January %s' % (last_year), self.stage_1.id): 1,
    ('February %s' % (last_year), self.stage_2.id): 1,
    ('March %s' % (last_year), self.stage_3.id): 1,
}
# add the previous year's stage_4 entries
for month in months[3:]:  # starting from April
    month_key = f"{month} {last_year}"
    project_expected_dict[(month_key, self.stage_4.id)] = 1
# add the current year's stage_4 entries
for i in range(current_month):
    month_key = f"{months[i]} {self.current_year}"
    project_expected_dict[(month_key, self.stage_4.id)] = 2
```